internal/util/grpcclient/client.go

```diff
@@ -368,6 +369,10 @@ func (c *ClientBase[T]) checkErr(ctx context.Context, err error) (needRetry, nee
 			// canceled or deadline exceeded
 			return true, c.needResetCancel()
 		}
+
+		if funcutil.IsGrpcErr(err, codes.Unimplemented) {
```

Comment:
We also need to check the interfaces return unimplemented error, introduce this in next PR?

Reply:
good catch

Reply:
here will return a Unrecoverable Error, but `IsGrpcError` cloud go deep into every internal error, and found the `Unimplemented` error code